crates/youki/src/observability.rs

```diff
@@ -73,7 +73,13 @@ where
     let log_level_filter = tracing_subscriber::filter::LevelFilter::from(level);
     let log_format = detect_log_format(config.log_format.as_deref())
         .with_context(|| "failed to detect log format")?;
-    let systemd_journald = if config.systemd_log {
+
+    #[cfg(debug_assertions)]
```

Comment:
Is this intended to be merged with this PR?

Reply:
I added this because it helped with debugging youki when running via podman, because a lot of logs were not getting displayed by podman. I can remove this in this PR and we can come back to this in the issue you have opened for the logs, this is not necessary for the rootless changes.

Reply:
I see. We can keep it.